lib/MusicBrainz/Server/Controller/Relationship/LinkAttributeType.pm

```diff
@@ -99,24 +101,38 @@ sub edit : Chained('load') RequireAuth(relationship_editor)
 
     my $form = $c->form( form => 'Admin::LinkAttributeType', init_object => $link_attr_type );
 
+    my $attribute_in_use = $c->model('LinkAttributeType')->in_use($link_attr_type->id);
+
+    if ($attribute_in_use) {
+        $form->field('free_text')->disabled(1);
+        if ($link_attr_type->creditable) {
+            $form->field('creditable')->disabled(1);
+        }
+    }
+
     if ($c->form_posted_and_valid($form)) {
+        my %values = map { $_->name => $_->value } $form->edit_fields;
         $c->model('MB')->with_transaction(sub {
-            $self->_insert_edit(
+            my $edit = $self->_insert_edit(
                 $c, $form,
                 edit_type => $EDIT_RELATIONSHIP_ATTRIBUTE,
                 entity_id => $link_attr_type->id,
-                new => { map { $_->name => $_->value } $form->edit_fields },
+                new => \%values ,
                 old => {
                     name => $link_attr_type->name,
                     description => $link_attr_type->description,
                     parent_id => $link_attr_type->parent_id,
                     child_order => $link_attr_type->child_order,
+                    creditable => $link_attr_type->creditable,
+                    free_text => $link_attr_type->free_text,
                 }
             );
-        });
 
-        $c->response->redirect($c->uri_for_action('relationship/linkattributetype/show', [ $link_attr_type->gid ]));
-        $c->detach;
+            if ($edit) {
+                $c->response->redirect($c->uri_for_action('relationship/linkattributetype/show', [ $link_attr_type->gid ]));
+                $c->detach;
```

Comment:
I see now, `detach` throws an exception (which is normal, that's how it works), which rolls back the transaction.  So we should leave the redirect code outside of `with_transaction`.  You can still declare `my $edit;` outside of `with_transaction` and do the assignment inside.

Reply:
(The detach shouldn't be needed in either case -- though I'd still move the redirect outside.)

Reply:
Thanks! Seems to work, would never have figured it out, sigh :)